portfolio/src/main/java/com/google/sps/data/BasketballCourtsDatastore.java

```diff
@@ -0,0 +1,102 @@
+package com.google.sps.data;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonElement;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import com.google.sps.data.BasketballCourt;
+
+/** Parses the basketball courts file and returns a list of basketball courts */
+public final class BasketballCourtsDatastore {
+  private static final Logger log = Logger.getLogger(BasketballCourtsDatastore.class.getName());
+
+  public static final String JSON_FILE_PATH =
+    "///home/riyanar/step/portfolio/src/main/webapp/WEB-INF/basketball-courts-nyc.json";
+  public static final String ACCESSIBLE_PROPERTY = "Accessible";
+  public static final String IS_ACCESSIBLE = "Y";
+  public static final String ID_PROPERTY = "Prop_ID";
+  public static final String NAME_PROPERTY = "Name";
+  public static final String LOCATION_PROPERTY = "Location";
+  public static final String NUMCOURTS_PROPERTY = "Num_of_Courts";
+  public static final String LAT_PROPERTY = "lat";
+  public static final String LON_PROPERTY = "lon";
+
+  private BasketballCourtsDatastore() {}
+
+  /** Returns a list of the accessible basketball courts. If there is a bad entry, it should be
+    ignored and the other entries should be returned. */
+  public static List<BasketballCourt> getAccessibleBasketballCourts() {
+
+    ImmutableList.Builder<BasketballCourt> accessibleCourts =
+      new ImmutableList.Builder<BasketballCourt>();
+
+    try {
+      String path = JSON_FILE_PATH;
+      BufferedReader bufferedReader = new BufferedReader(new FileReader(path));
```

Comment:
your reader must be freed when you are done reading. Expecting to see "try-with-resource" here. 
```
try(BufferedReader bufferedReader = new [...]){


}catch(...){
   logError
}
```